src/lib/Syntax.hs

```diff
@@ -766,6 +810,10 @@ instance Eq Atom where
   DataCon def params con args == DataCon def' params' con' args' =
     def == def' && params == params' && con == con' && args == args'
   TypeCon def params == TypeCon def' params' = def == def' && params == params'
+  Record lr == Record lr' = lr == lr'
+  Variant lr l i v == Variant lr' l' i' v' = (lr, l, i, v) == (lr', l', i', v')
+  RecordTy lr == RecordTy lr' = lr == lr'
+  VariantTy lr == VariantTy lr' = lr == lr'
```

Comment:
nit: can you please ad some alignment, because right not this is very hard to read